packages/atlas-service/src/components/atlas-signin.tsx

```diff
@@ -1,16 +1,27 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useRef } from 'react';
 import { Provider, useDispatch } from 'react-redux';
 import AISignInModal from './ai-signin-modal';
+import type { AtlasServiceStore } from '../store/atlas-signin-store';
 import { getStore } from '../store/atlas-signin-store';
 import { restoreSignInState } from '../store/atlas-signin-reducer';
+import { ConfirmationModalArea } from '@mongodb-js/compass-components';
 
-function Modal() {
-  const dispatch = useDispatch();
+export const Modal: React.FunctionComponent<{
+  restoreStateOnMount?: boolean;
```

Comment:
When wouldn't we want to restore the state on mount? Is this mostly for testing?

Reply:
Yes, just for testing